src/Sylius/Behat/Context/Setup/ProductContext.php

```diff
@@ -37,30 +45,57 @@ class ProductContext implements Context
     private $productFactory;
 
     /**
-     * @param RepositoryInterface $productRepository
      * @param SharedStorageInterface $sharedStorage
+     * @param RepositoryInterface $productRepository
+     * @param RepositoryInterface $taxCategoryRepository
      * @param FactoryInterface $productFactory
      */
-    public function __construct(RepositoryInterface $productRepository, SharedStorageInterface $sharedStorage, FactoryInterface $productFactory)
-    {
-        $this->productRepository = $productRepository;
+    public function __construct(
+        SharedStorageInterface $sharedStorage,
+        RepositoryInterface $productRepository,
+        RepositoryInterface $taxCategoryRepository,
+        FactoryInterface $productFactory
+    ) {
         $this->sharedStorage = $sharedStorage;
+        $this->productRepository = $productRepository;
+        $this->taxCategoryRepository = $taxCategoryRepository;
         $this->productFactory = $productFactory;
     }
 
     /**
-     * @Given /^store has a product "([^"]*)" priced at "(€|£|\$)([^"]*)"$/
+     * @Transform /^product "([^"]+)"$/
+     * @Transform /^"([^"]+)" product$/
+     */
+    public function castProductNameToProduct($productName)
+    {
+        if (null === $product = $this->productRepository->findOneBy(array('name' => $productName))) {
+            throw new \InvalidArgumentException('Product with name "'.$product.'" does not exist');
+        }
+
+        return $product;
+    }
+
+    /**
+     * @Given /^store has a product "([^"]+)" priced at "(€|£|\$)([^"]+)"$/
```

Comment:
Are we using the currency? If not, we can disable matching on that group `(?:€|£|\$)` or name the other parameters like: 
```regexp
/^store has a product "(?P<productName>[^"]+)" priced at "(€|£|\$)(?P<price>[^"]+)"$/
```